File: source_funcs/go_transcoder/PAIR_WITH_GIVEN_PRODUCT_SET_1_FIND_IF_ANY_PAIR_EXISTS/PAIR_WITH_GIVEN_PRODUCT_SET_1_FIND_IF_ANY_PAIR_EXISTS.go
package main

func f_gold(arr []int, n int, x int) bool {
	for i := int(0); i < n-1; i++ {
		for j := int(i + 1); i < n; i++ {
			if arr[i]*arr[j] == x {
				return true
			}
		}
	}
	return false
}

File: rust_funcs/go_transcoder/PAIR_WITH_GIVEN_PRODUCT_SET_1_FIND_IF_ANY_PAIR_EXISTS/PAIR_WITH_GIVEN_PRODUCT_SET_1_FIND_IF_ANY_PAIR_EXISTS.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}



fn f_gold(arr: [i32;2], n: i32, x: i32) -> i32 {
    let mut i = 0;
    while i < n - 1 {
        let mut j = i + 1; 
        while j < n {
            if arr[i as usize] as f32 * arr[j as usize] as f32 == x as f32{
                return 1;
            }
            j += 1;
        }
        i += 1;
    }
   0
}
